src/coerce-functions.js

```diff
@@ -15,7 +15,14 @@ export const coerceFunctions = {
     return !!a;
   },
   date(a) {
-    return new Date(a);
+    // Invalid date instances are quite problematic
+    // so we need to deal with it properly by default
+    if (val === null || val === undefined) {
+      return null;
+    }
+    const d = new Date(val);
+    const t = d.getTime(); // to deal with invalid date
+    return t === t ? d : null;

```

Comment:
is this `t === t` an optimization compared to `!isNaN(t)`, or why do you use this weird syntax, that only makes sense in JavaScript?

Reply:
Yes it is used to specifically deal with NaN value. As `NaN === NaN` is `false`.

What a spec

Reply:
But does it perform better than using `isNaN`?

Reply:
Thats interesting question. Normally i would say yes but V8 is smart and I believe isNaN could be faster ( from another v8 engineer tweet i saw different between `o === o` and `Object.is(o, o)` is 2 us

Reply:
@bigopon 
> As `NaN === NaN` is `false`.

Yeah i know, after figuring that out I started using TsLint ;)

My comment+question was because of readability & preformance. IMHO if `t === t` is not preforming (much) better than `!isNaN(t)`, then i'd vote for better readability :)

Reply:
I see. Cant commit on phone 😁 Will change

Reply:
@atsu85 thanks for the suggestion. I just keep doing those things 😄